### PR TITLE
DYN-3243: HTML Sanitizer fixes

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

HTML Sanitizer fixes

This pull request does:
* allow html content with a `body` to Sanitize as a document (e.g. `head` tags are allowed)
* allow `meta` tags
* allow `style` tags
* allow `content` attributes
* allow `http-equiv` attributes
* allow `src` css properties
* stop consider comments as a content change

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@mjkkirschner

### CODE REVIEW

File: src/Tools/Md2Html/Md2Html.cs [hunk 1]
```diff
@@ -97,8 +97,16 @@ private static void ConvertRelativeLocalImagePathsToAbsolute(string mdFilePath,
         /// <returns>return sanitized content string or an empty string if no sanitizing happened</returns>
         internal static string Sanitize(string content)
         {
+            string output;
             HtmlSanitizer.ContentWasUpdated = false;
-            var output = HtmlSanitizer.Sanitize(content);
+            if (content.Contains(@"<body "))
+            {
+                output = HtmlSanitizer.SanitizeDocument(content);
```
Comment (mjkkirschner) [inline src/Tools/Md2Html/Md2Html.cs:104]: whats the difference between these two?

**Comment by mjkkirschner** [comment]
@sm6srw will the current sanitization settings allow images embedded using data like this:
https://github.com/DynamoDS/Dynamo/blob/master/src/DocumentationBrowserViewExtension/Docs/FileNotFound.html#L57 ?

File: src/Tools/Md2Html/Md2Html.cs (hunk 1 above)
Comment (sm6srw) [reply src/Tools/Md2Html/Md2Html.cs:104]: This one sanitize a document, e.g. you have a body with an optional head. The other one is for sanitizing fragments. head is ignored for fragments and that was the problem here.

**Comment by sm6srw** [comment]
> @sm6srw will the current sanitization settings allow images embedded using data like this:
> https://github.com/DynamoDS/Dynamo/blob/master/src/DocumentationBrowserViewExtension/Docs/FileNotFound.html#L57 ?

Yes, I think so but I will give it a test.

**Review by mjkkirschner — APPROVED** [review]

**Comment by mjkkirschner** [comment]
This looks good @sm6srw but as we have seen - we should probably beef up our test cases.

**Comment by mjkkirschner** [comment]
@sm6srw tests all pass - feel free to merge when you think it's ready.

**Pull request merged**